src/lib/fountain/concurrent_fountain_decoder_sink.h

```diff
@@ -6,12 +6,11 @@
 #include "concurrentqueue/concurrentqueue.h"
 #include <mutex>
 
-template <typename OUTSTREAM>
 class concurrent_fountain_decoder_sink
 {
 public:
-	concurrent_fountain_decoder_sink(std::string data_dir, unsigned chunk_size)
-	    : _decoder(data_dir, chunk_size)
+	concurrent_fountain_decoder_sink(unsigned chunk_size, const std::function<void(const std::string&, const std::vector<uint8_t>&)>& on_store=nullptr)
+		: _decoder(chunk_size, on_store)
```

Comment:
Every time QtCreator messes with whitespace because my settings have subtly changed in the past 5 years, I get a little closer to adding clang-format to the project

Reply:
(I've been planning on doing it. Though I hate what they do to my precious anonymous namespaces)